src/c_ffi.rs

```diff
@@ -0,0 +1,33 @@
+use libc::{c_char};
+use std::ffi::{CString};
+
+use super::Trie;
+use super::trie_common::TrieCommon;
+
+ffi_fn! {
+    fn radix_trie_create()-> *const Trie<CString, usize>{
+        let trie = Trie::<CString, usize>::new();
+        return Box::into_raw(Box::new(trie));
+    }
+}
+
+ffi_fn! {
+    fn radix_trie_free(trie_ptr: *const Trie<CString, usize>){
+        unsafe { Box::from_raw(trie_ptr as *mut Trie<CString, usize>); }
+    }
+}
+
+ffi_fn! {
+    fn radix_trie_insert(trie_ptr:*const Trie<CString, usize>, key_ptr:*const c_char, value:usize){
+        let trie = unsafe { &mut *(trie_ptr as *mut Trie<CString, usize>) };
```

Comment:
I think the `trie_ptr` argument to this function should be `*mut Trie<CString, usize>`, at least to convey the intent that it's a mutable pointer (and to allow us to remove some of the casting gunk)

Reply:
We also need to run `rustfmt` over the whole project before the build will pass, you can see the `.travis.yml` for how to set it up

Thanks!

Reply:
Fixed

Reply:
Thanks! You can remove the cast `(trie_ptr as *mut Trie<CString, usize>)` too

Reply:
Good catch!  resolved